src/test/groovy/org/prebid/server/functional/tests/BidExpResponseSpec.groovy

```diff
@@ -0,0 +1,346 @@
+import org.prebid.server.functional.util.PBSUtils
+
+class BidExpResponseSpec extends BaseSpec {
+
+    private static def hostBannerTtl = PBSUtils.randomNumber
+    private static def hostVideoTtl = PBSUtils.randomNumber
+    private static def cacheTtlService = pbsServiceFactory.getService(['cache.banner-ttl-seconds': hostBannerTtl as String,
+                                                                       'cache.video-ttl-seconds' : hostVideoTtl as String])
+
+    def "PBS auction should resolve bid.exp from response that is set by the bidder’s adapter"() {
+        given: "Default basicResponse with exp"
+        def bidResponseExp = PBSUtils.randomNumber
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].exp = bidResponseExp
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain exp data"
+        assert response.seatbid.bid.first.exp == [bidResponseExp]
+
+        and: "PBS should not call PBC"
+        assert !prebidCache.getRequestCount(bidRequest.imp.first.id)
+
+        where:
+        bidRequest << [BidRequest.defaultBidRequest, BidRequest.defaultVideoRequest]
+    }
+
+    def "PBS auction should resolve bid.exp from response and send it to cache when it set by the bidder’s adapter and cache enabled for request"() {
+        given: "BidRequest with enabled cache"
+        bidRequest.enableCache()
+
+        and: "Default basic bid with exp"
+        def bidResponseExp = PBSUtils.randomNumber
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].exp = bidResponseExp
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain exp data"
+        assert response.seatbid.bid.first.exp == [bidResponseExp]
+
+        and: "PBS should call PBC"
+        def cacheRequests = prebidCache.getRecordedRequests(bidRequest.imp.first.id)
+        assert cacheRequests.puts.first.first.ttlseconds == bidResponseExp
+
+        where:
+        bidRequest << [BidRequest.defaultBidRequest, BidRequest.defaultVideoRequest]
+    }
+
+    def "PBS auction should resolve exp from request.imp[].exp when it have value"() {
+        given: "Default basic bidRequest with exp"
+        def bidRequestExp = PBSUtils.randomNumber
+        bidRequest.tap {
+            imp.first.exp = bidRequestExp
+        }
+
+        and: "Set bidder response without exp"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        bidder.setResponse(bidRequest.id, bidResponse)
```

Comment:
It would be great to see explicitly `without exp`, since in the future somebody can change the default bid response and add `exp` field. by default